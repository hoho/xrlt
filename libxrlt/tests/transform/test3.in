id:0, type:0, last:0, error: 0, data:
id:0, type:0, last:0, error: 0, data:
id:1, type:2, last:0, error: 0, data:"hello
id:1, type:2, last:1, error: 0, data: world"
id:0, type:0, last:0, error: 0, data:
id:3, type:2, last:1, error: 0, data:some
id:2, type:2, last:1, error: 0, data:<param><value>awe</value></param>
id:0, type:0, last:0, error: 0, data:
id:4, type:2, last:1, error: 0, data:{"this": "is", "pretty": "amazing"}
id:0, type:0, last:0, error: 0, data:
