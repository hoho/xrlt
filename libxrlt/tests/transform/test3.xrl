<?xml version="1.0"?>
<xrl:requestsheet xmlns:xrl="http://xrlt.net/Transform">
    <xrl:response>
        <xrl:variable name="href">
            <xrl:include>
                <xrl:href>/test/href/1</xrl:href>
                <xrl:type>json</xrl:type>
                <xrl:success select="/" />
            </xrl:include>
        </xrl:variable>

        <xrl:variable name="par1">
            <xrl:include>
                <xrl:href>/test/href/2</xrl:href>
                <xrl:type>xml</xrl:type>
                <xrl:success select="/param/value" />
            </xrl:include>
        </xrl:variable>

        <xrl:variable name="par2">
            <xrl:include>
                <xrl:href>/test/href/3</xrl:href>
                <xrl:type>text</xrl:type>
            </xrl:include>
        </xrl:variable>

        <hi>yo</hi>

        <xrl:include>
            <xrl:href select="$href" />
            <xrl:type>json</xrl:type>
            <xrl:with-param name="p" select="concat($par1, $par2)" />
        </xrl:include>

        <bye>yep</bye>
    </xrl:response>
</xrl:requestsheet>
