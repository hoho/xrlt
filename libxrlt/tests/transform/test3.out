XRLT_STATUS_CHUNK
chunk: yo
XRLT_STATUS_SUBREQUEST
sr id: 1
sr method: GET
sr type: JSON
sr url: /test/href/1
sr query: (null)
sr body: (null)
XRLT_STATUS_WAITING
XRLT_STATUS_SUBREQUEST
sr id: 2
sr method: GET
sr type: XML
sr url: /test/href/2
sr query: (null)
sr body: (null)
XRLT_STATUS_SUBREQUEST
sr id: 3
sr method: GET
sr type: TEXT
sr url: /test/href/3
sr query: (null)
sr body: (null)
XRLT_STATUS_WAITING
XRLT_STATUS_SUBREQUEST
sr id: 4
sr method: GET
sr type: JSON
sr url: hello world
sr query: p=awesome
sr body: (null)
XRLT_STATUS_WAITING
XRLT_STATUS_CHUNK
chunk: is
chunk: amazing
chunk: yep
XRLT_STATUS_DONE
